volume_zarr.py

```diff
@@ -0,0 +1,636 @@
+        the corners of the dataset.
+        """
+        gmin = np.array([0, 0, 0], dtype=np.int32)
+        gmax = np.array(self.sizes, dtype=np.int32) - 1
+        return np.array((gmin, gmax))
+
+    def loadData(self, project_view):
+        self.active_project_views.add(project_view)
+
+    def unloadData(self, project_view):
+        self.active_project_views.discard(project_view)
+        self.data.clear_cache()
+
+    def createTransposedData(self):
+        pass
+
+    def ijkToTransposedIjk(self, ijk, direction):
+        i,j,k = ijk
+        if direction == 0:
+            return (j,k,i)
+        else:
+            return (i,k,j)
+
+    def transposedIjkToIjk(self, ijkt, direction):
+        it,jt,kt = ijkt
+        if direction == 0:
+            return (kt,it,jt)
+        else:
+            return (it,kt,jt)
+
+    def transposedIjkToGlobalPosition(self, ijkt, direction):
+        return self.transposedIjkToIjk(ijkt, direction)
+
+    def globalPositionsToTransposedIjks(self, gpoints, direction):
+        if direction == 0:
+            return gpoints[:,(1,2,0)]
+        else:
+            return gpoints[:,(0,2,1)]
+
+    def transposedIjksToGlobalPositions(self, ijkts, direction):
+        if direction == 0:
+            return ijkts[:,(2,0,1)]
+        else:
+            return ijkts[:,(0,2,1)]
+
+    def getGlobalRanges(self):
+        arr = []
+        for i in range(3):
+            # Note we reverse the indices here
+            arr.append([0, self.data.shape[2 - i]])
+        return arr
+
+    def globalAxisFromTransposedAxis(self, axis, direction):
+        if direction == 0:
+            return (axis + 1) % 3
+        else:
+            return (0,2,1)[axis]
+
+    def getSlice(self, axis, ijkt, direction):
+        """ijkt are the coordinates of the center point in transposed space
+        """
+        (it, jt, kt) = self.transposedIjkToIjk(ijkt, direction)
+        islice = slice(
+            max(0, it - self.max_width), 
+            min(self.data.shape[2], it + self.max_width + 1),
+            None,
+        )
+        jslice = slice(
+            max(0, jt - self.max_width), 
+            min(self.data.shape[1], jt + self.max_width + 1),
+            None,
+        )
+        kslice = slice(
+            max(0, kt - self.max_width), 
+            min(self.data.shape[0], kt + self.max_width + 1),
+            None,
+        )
+
+        if axis == 1:
+            return self.data[kt, jslice, islice]
+        elif axis == 2:
+            return self.data[kslice, jt, islice]
+        elif axis == 0:
+            return self.data[kslice, jslice, it].T
+        raise ValueError
+
+    def ijIndexesInPlaneOfSlice(self, axis):
+        return ((1,2), (0,2), (0,1))[axis]
+
+    def ijInPlaneOfSlice(self, axis, ijkt):
+        #inds = self.ijIndexesInPlaneOfSlice(axis)
+        #return(ijkt[inds[0]], ijkt[inds[1]])
+        return (self.max_width, self.max_width)
```

Comment:
```suggestion
        inds = self.ijIndexesInPlaneOfSlice(axis)
        x,y = ijkt[inds[0]], ijkt[inds[1]]
        return (min(x, self.max_width), min(y, self.max_width))
```

Reply:
Not sure if I am doing this correctly?  I'm new at using github for code review.  Let me know if you prefer a different approach.  Anyway, my change to ijInPlaneOfSlice is to fix an alignment problem at the left and top sides of the volume.